src/js/actions/libraries.js

```diff
@@ -514,31 +520,34 @@ define(function (require, exports) {
     applyLayerStyle.transfers = [layerActions.resetLayers];
 
     /**
-     * Applies the given character style element to the active layers
+     * Applies the given character style element to the selected text layers
      *  - Gets the path from primary representation of the asset
      *  - Uses textLayer adapter call to apply the style data
      *
-     * @param {AdobeLibraryElement} element [description]
+     * @param {AdobeLibraryElement} element
      *
      * @return {Promise}
      */
     var applyCharacterStyle = function (element) {
         var appStore = this.flux.store("application"),
-            currentDocument = appStore.getCurrentDocument();
+            currentDocument = appStore.getCurrentDocument(),
+            selectedLayers = currentDocument ? currentDocument.layers.selected : new Immutable.List(),
+            textLayers = selectedLayers.filter(function (l) { return l.isTextLayer(); });
 
-        if (!currentDocument) {
+        if (!currentDocument || textLayers.isEmpty()) {
             return Promise.resolve();
         }
 
         var representation = element.getPrimaryRepresentation(),
             styleData = representation.getValue("characterstyle", "data"),
-            layerRef = textLayerAdapter.referenceBy.current,
+            textLayerIDs = collection.pluck(textLayers, "id"),
+            layerRef = textLayerIDs.map(textLayerAdapter.referenceBy.id).toArray(),
             applyObj = textLayerAdapter.applyTextStyle(layerRef, styleData);
 
-        return descriptor.playObject(applyObj)
+        return layerActionsUtil.playSimpleLayerActions(currentDocument, textLayers, applyObj, true)
```

Comment:
I have to make the changes here to only apply char style to text layers to address an issue (Set command is not available) when you select all layers (including the art board).